### DISCUSSION TITLE
Accessing the current local in form event listener

### DISCUSSION THREAD

**Discussion by simonbedard12** [statamic/cms#4590] [Troubleshooting] [answered]
Hi

## Environment 
This is my current Statamic Setup: `php please support:details`

```
Statamic 3.2.4 Pro
Laravel 8.52.0
PHP 7.4.21
No addons installed
```
## Problem

The `Site::current()` function will always return the default local in a form event listener like `Statamic\Events\SubmissionCreated`

The `Site::current()` function return the current local base on the URI: https://github.com/statamic/cms/blob/3.2/src/Sites/Sites.php (Line 51)

Since I'm in a form event the current URI is always starting with `/!/form/.....` 

## Working solution

A working solution will be to add a field in the form and use the submission event to retreave the data.
I feel like that this is not a great solution. 

## Question
What will be the best solution to get the current local in a form event class.  

Thanks,
Simon

**Comment by jelleroorda** [comment]
Hi Simon, from what I see it's not possible at this moment. The site would probably need to be passed inside [the SubmissionCreated event here](https://github.com/statamic/cms/blob/3.2/src/Http/Controllers/FormController.php#L68) so you can use the site's locale.

**Comment by jasonvarga** [comment, marked as answer]
You could do this:

```php
$site = Statamic\Facades\Site::findByUrl(url()->previous())
```

It probably won't work if you're using a queued listener though.

Ideally, we'd pass the site in the event, like @jelleroorda said.

**Reply by duncanmcclean** [reply]
One possibility would also be to have a hidden input in the form which stores the handle of the related site.

```html
<input type="hidden" name="_site" value="{{ site:handle }}">
```

Which you can then use in your listener...

```php
$event->submission->get('_site');
```

**Reply by simonbedard12** [reply]
I tried both solutions and they are working has they should.
For future readers, choose the one that suits your needs.
